samples/cpp/stitching.cpp

```diff
@@ -80,58 +80,54 @@ void printUsage()
 {
     cout <<
         "Rotation model images stitcher.\n\n"
-        "stitching img1 img2 [...imgN]\n\n"
+        "stitching -n=N img1 img2 [...imgN]\n\n"
         "Flags:\n"
-        "  --try_use_gpu (yes|no)\n"
+        "  --try_use_gpu=(yes|no)\n"
         "      Try to use GPU. The default value is 'no'. All default values\n"
         "      are for CPU mode.\n"
-        "  --output <result_img>\n"
+        "  --output=<result_img>\n"
         "      The default is 'result.jpg'.\n";
 }
 
 
 int parseCmdArgs(int argc, char** argv)
 {
-    if (argc == 1)
+    cv::CommandLineParser parser(argc, argv, "{help||}{try_use_gpu|no|}{output|result.jpg|}{n|0|}");
+    if (parser.has("help"))
     {
         printUsage();
         return -1;
     }
-    for (int i = 1; i < argc; ++i)
+    if (parser.get<string>("try_use_gpu") == "no")
+        try_use_gpu = false;
+    else if (parser.get<string>("try_use_gpu") == "yes")
+        try_use_gpu = true;
+    else
     {
-        if (string(argv[i]) == "--help" || string(argv[i]) == "/?")
+        cout << "Bad --try_use_gpu flag value\n";
+        return -1;
+    }
+    result_name = parser.get<string>("output");
+    int n = parser.get<int>("n");
+    if (!parser.check())
+    {
+        printUsage();
+        return -1;
+    }
+    if (n == 0)
+    {
+        printUsage();
+        return -1;
+    }
+    for (int i = 0; i < n ; i++)
+    {
+        Mat img = imread(parser.get<string>(i));
```

Comment:
It will not work. Currently CommandLineParser does not support variable positional arguments count, so I'd recommend to revert the whole file.